pkg/codegen/hcl2/model/type_opaque.go

```diff
@@ -34,33 +35,74 @@ type OpaqueType struct {
 	s string
 }
 
-// The set of opaque types, indexed by name.
-var opaqueTypes = map[string]*OpaqueType{}
+// A global type map for built-in types, e.g.: "bool"
+var globalTypes = make(map[string]*OpaqueType)
 
-// GetOpaqueType fetches the opaque type for the given name.
-func GetOpaqueType(name string) (*OpaqueType, bool) {
-	t, ok := opaqueTypes[name]
-	return t, ok
+type OpaqueTypeMap interface {
```

Comment:
An interface was chosen here to ensure that anything that constructs this, does so by pointer ref and thus the mutex is copied by reference and not value.